apps/local_laplacian/local_laplacian_generator.cpp

```diff
@@ -124,17 +117,16 @@ class LocalLaplacian : public Halide::Generator<LocalLaplacian> {
             }
             outGPyramid[0]
                 .compute_at(output, y).vectorize(x, 8);
-            for (int j = 5; j < J; j++) {
+            for (int j = 5; j < pyramid_levels; j++) {
                 inGPyramid[j].compute_root();
                 gPyramid[j].compute_root().parallel(k);
                 outGPyramid[j].compute_root();
             }
         }
-
-        return output;
     }
 private:
     Var x, y, c, k;
+    Func remap, gray, inGPyramid[maxJ], gPyramid[maxJ], outGPyramid[maxJ];
```

Comment:
The design goal here is: You shouldn't have to be aware of a feature until it solves a problem that you have. When you do become aware of it, you shouldn't need to restructure all the code you've written so far to make use of it.

In this specific instance it means that people can just put everything in generate(), and when they need a ScheduleParam they can add it as a member without having to tease apart their code into generate() and schedule(). This particular demo app can ignore deferred scheduling. camera_pipe would use it.